mesh_tensorflow/signal_ops.py

```diff
@@ -30,48 +78,24 @@ class FFT3DOperation(mtf.Operation):
     name: A name for the operation (optional).
 
   Returns:
-    A Tensor of shape `input.shape[:-3] + freq_dims`.
+    A Tensor of shape `input.shape[:-3] + freq_dims[1] + freq_dims[2] + freq_dims[0]`.
   """
-  def __init__(self, input,  freq_dims, name=None):
-    super(FFT3DOperation, self).__init__([input], name=name or "FFT3D")
-    self._freq_dims = freq_dims
-    self._output_shape = mtf.Shape(input.shape[:-3]+[freq_dims[1], freq_dims[2], freq_dims[0]])
-    self._outputs = [mtf.Tensor(self, mtf.Shape(self._output_shape), input.dtype)]
+  def __init__(self, inputs,  dims, name=None):
+    super(FFT3DOperation, self).__init__(inputs, dims, inverse=False, name=name)
+
+  def _transpose(self, mesh_impl, split_axes, slices, naxes):
```

Comment:
While we are at it.... we may as well store that function directly in the Base class with a switch dependending on the self.inverse flag, and then we don't even need that children class

Reply:
I was thinking that since it's quite a "big" switch it was better suited for 2 different classes.
I also thought since the 2 classes have different return shapes it was best to clearly have 2 separate ones.

At this point it's clearly a matter of taste I would say so I'll let you decide and make the change if you want